src/Network/Network/ApplicationGateway/SslCertificate/SetAzureApplicationGatewaySslCertificateCommand.cs

```diff
@@ -40,7 +40,9 @@ public override void ExecuteCmdlet()
                 throw new ArgumentException("Ssl certificate with the specified name does not exist");
             }
             
-            X509Certificate2 cert = new X509Certificate2(this.CertificateFile, this.Password, X509KeyStorageFlags.Exportable);
+            if (this.CertificateFile != null) {
```

Comment:
We need a test to protect us from regressing after this fix goes in.  Also, need a change in the network changelog, per the PR checklist

Reply:
@markcowl Do we have the ability to mock keyvault responses now for data plane ?

Reply:
No, but we can mock the keyvault client altogether. there should at least be a test so we can validate the network portion